commit 22958c1b2d602fa1bc301a7f717dc9632dfce3c2 (refs/changes/72/5272/2)
Author: Josh Wilsdon <josh@wilsdon.ca>
Date:   2018-12-20T11:05:05-08:00 (10 months ago)
    
    TRITON-1053 CNAPI should not crash when someone calls `GET /servers/`
    Reviewed by: Orlando Vazquez <orlando@joyent.com>
    Reviewed by: Trent Mick <trentm@gmail.com>
    Approved by: Orlando Vazquez <orlando@joyent.com>
    Approved by: Trent Mick <trentm@gmail.com>

diff --git a/lib/endpoints/index.js b/lib/endpoints/index.js
index 1ff4b46..318b691 100644
--- a/lib/endpoints/index.js
+++ b/lib/endpoints/index.js
@@ -77,6 +77,18 @@ function ensureConnectionTimeout(opts, req, res, next) {
     next();
 }
 
+function isUuid(candidate) {
+    // borrowed from assert-plus
+    /* JSSTYLED */
+    var UUID_REGEXP = /^[a-fA-F0-9]{8}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{4}-[a-fA-F0-9]{12}$/;
+
+    if (typeof (candidate) === 'string' && UUID_REGEXP.test(candidate)) {
+        return true;
+    }
+
+    return false;
+}
+
 
 /* BEGIN JSSTYLED */
 /**
@@ -103,9 +115,14 @@ function ensureServerRunning(opts, req, res, next) {
     });
 }
 
-
 function prepopulateServer(opts, req, res, next) {
-    assert.string(req.params.server_uuid, 'server_uuid');
+    assert.object(req, 'req');
+    assert.object(req.params, 'req.params');
+
+    if (!isUuid(req.params.server_uuid)) {
+        next(new restify.InvalidArgumentError('Invalid server_uuid'));
+        return;
+    }
 
     ModelServer.get(
         req.params.server_uuid,
@@ -130,6 +147,13 @@ function prepopulateServer(opts, req, res, next) {
 
 function prepopulateVm(opts, req, res, next) {
     assert.string(req.params.uuid, 'uuid');
+    assert.object(req, 'req');
+    assert.object(req.params, 'req.params');
+
+    if (!isUuid(req.params.uuid)) {
+        next(new restify.InvalidArgumentError('Invalid vm_uuid'));
+        return;
+    }
 
     var uuid = req.params.uuid;
 
diff --git a/package.json b/package.json
index 63010f7..5a87db4 100644
--- a/package.json
+++ b/package.json
@@ -1,7 +1,7 @@
 {
   "name": "cnapi",
   "description": "SmartDataCenter Compute Node API",
-  "version": "1.18.0",
+  "version": "1.18.1",
   "author": "Joyent (joyent.com)",
   "private": true,
   "dependencies": {
